src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/service/LimitedPartnershipService.java

```diff
@@ -22,12 +25,19 @@ public LimitedPartnershipService(LimitedPartnershipMapper mapper, LimitedPartner
         this.repository = repository;
     }
 
-    public String createLimitedPartnership(LimitedPartnershipSubmissionDto limitedPartnershipSubmissionDto, String requestId, String userId) {
+    public String createLimitedPartnership(Transaction transaction, LimitedPartnershipSubmissionDto limitedPartnershipSubmissionDto, String requestId, String userId) {
         ApiLogger.debug("Called createLimitedPartnership(...)");
 
         LimitedPartnershipSubmissionDao dao = mapper.dtoToDao(limitedPartnershipSubmissionDto);
         dao.setCreatedAt(LocalDateTime.now());
         dao.setUserId(userId);
+
+        // Create the self-link
+        String selfLink = String.format("/transactions/%s/limited-partnership-submissions/%s", transaction.getId(), dao.getId());
```

Comment:
dao.getId() will be null at this point as the mongo record has not been created. The record should be updated with the self link after it has been inserted.

Reply:
revised the flow to enable update of self link after insertion

Reply:
The link is being added to the DAO but it's not being saved to the database. Test any change locally to ensure that it is working as required.

Reply:
if self-link is created and set in the dao before the dao is inserted into the repository, this ensures that the dao with the self-link is saved to the database.

Reply:
The self link includes the submission ID. This is generated when the record is created in Mongo. So the record (DAO) needs to be inserted into the repository first, then updated with the self link and saved again.

Reply:
have tested changes locally links now written to database after update with self link. see on [JIRA](https://companieshouse.atlassian.net/browse/LP-219)